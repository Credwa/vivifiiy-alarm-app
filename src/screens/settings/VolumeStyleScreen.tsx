import Background from '@/components/Background';
import VivText from '@/components/VivText';
import Colors from '@/constants/Colors';
import { AntDesign, Ionicons } from '@expo/vector-icons';
import React, { useState } from 'react';
import { Pressable, StyleSheet, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import useStore from '@/store/settings';
import { SettingsTabParamList } from '@/types';
import { StackScreenProps } from '@react-navigation/stack';
import { widthPercentageToDP as wp } from 'react-native-responsive-screen';
import VivCard from '@/components/VivCard';
import SettingCardItem from '@/components/SettingCardItem';
import { resize } from '@/utils';

export default function VolumeStyleScreen({
  navigation
}: StackScreenProps<SettingsTabParamList, 'VolumeStyleScreen'>) {
  const [selected, setSelected] = useState(useStore.getState().getSetting('volumeStyle'));
  const setVolumeStyle = useStore((state) => state.setSetting);

  const updatedSelected = (volumeStyle: string) => {
    setSelected(volumeStyle);
    setVolumeStyle('volumeStyle', volumeStyle);
  };

  return (
    <Background>
      <SafeAreaView style={styles.container}>
        <View>
          <View style={styles.header}>
            <Pressable onPress={navigation.goBack} style={styles.backButton}>
              {({ pressed }) => (
                <AntDesign
                  name="arrowleft"
                  size={resize(28, 24, 44)}
                  color={pressed ? Colors.greyLight3 : Colors.greyLight1}
                />
              )}
            </Pressable>
            <VivText
              fontName={resize('Title6', 'Body', 'Title3')}
              color={Colors.greyLight1}
              style={{ marginLeft: -15 }}
            >
              Volume style
            </VivText>
            <Pressable onPress={navigation.goBack} style={{ opacity: 0 }}>
              {({ pressed }) => (
                <AntDesign name="arrowleft" size={28} color={pressed ? Colors.greyLight3 : Colors.greyLight1} />
              )}
            </Pressable>
          </View>
        </View>
        <VivCard
          style={styles.card}
          headerChildren={
            <VivText fontName={resize('Headline', 'Headline', 'Title5')} color={Colors.greyLight1}>
              Alarm settings
            </VivText>
          }
        >
          <SettingCardItem
            title="Progressive"
            horizontalRule
            itemRight="icon"
            onCardItemPress={() => {
              updatedSelected('Progressive');
            }}
            rightIcon={
              selected === 'Progressive' ? <Ionicons name="ios-checkmark" size={28} color={Colors.greyLight2} /> : null
            }
          />

          <SettingCardItem
            title="Adaptive"
            horizontalRule
            itemRight="icon"
            onCardItemPress={() => {
              updatedSelected('Adaptive');
            }}
            rightIcon={
              selected === 'Adaptive' ? <Ionicons name="ios-checkmark" size={28} color={Colors.greyLight2} /> : null
            }
          />

          <SettingCardItem
            title="Standard"
            onCardItemPress={() => {
              updatedSelected('Standard');
            }}
            itemRight="icon"
            rightIcon={
              selected === 'Standard' ? <Ionicons name="ios-checkmark" size={28} color={Colors.greyLight2} /> : null
            }
          />
        </VivCard>
      </SafeAreaView>
    </Background>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginTop: 25
  },
  backButton: {
    height: 44,
    width: 44,
    alignContent: 'flex-start',
    alignItems: 'flex-start',
    justifyContent: 'center'
  },
  card: {},
  header: {
    flexDirection: 'row',
    height: 44,
    marginBottom: 50,
    width: wp('91%'),
    alignItems: 'center',
    justifyContent: 'space-between',
    alignContent: 'space-around'
  },
  separator: {
    marginVertical: 30,
    height: 0.3,
    width: '80%'
  }
});
