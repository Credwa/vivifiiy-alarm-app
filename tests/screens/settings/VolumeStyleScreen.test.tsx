import VolumeStyleScreen from '@/screens/settings/VolumeStyleScreen';
import { AntDesign, Ionicons } from '@expo/vector-icons';
import * as Font from 'expo-font';
import React from 'react';
import renderer from 'react-test-renderer';
jest.useFakeTimers();

describe('<VolumeStyleScreen />', () => {
  beforeEach(async () => {
    await Font.loadAsync({
      ...AntDesign.font,
      ...Ionicons.font
    });
  });

  it(`renders correctly`, async () => {
    const mockNav: any = { goBack: jest.fn() };
    const mockRoute: any = { params: {} };
    let tree = await renderer.create(<VolumeStyleScreen navigation={mockNav} route={mockRoute} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
