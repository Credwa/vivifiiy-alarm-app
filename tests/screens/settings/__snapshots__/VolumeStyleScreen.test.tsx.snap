// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VolumeStyleScreen /> renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 20,
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RNCSafeAreaView
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 25,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignContent": "space-around",
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "space-between",
            "marginBottom": 50,
            "width": 682.5,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "flex-start",
              "alignItems": "flex-start",
              "height": 44,
              "justifyContent": "center",
              "width": 44,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#CEE1F2",
                  "fontSize": 28,
                },
                undefined,
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#869DB2",
              },
              Object {
                "fontFamily": "Nunito_400Regular",
                "fontSize": 20,
              },
              Object {
                "color": "#CEE1F2",
              },
              Object {
                "marginLeft": -15,
              },
            ]
          }
        >
          Volume style
        </Text>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 0,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#CEE1F2",
                  "fontSize": 28,
                },
                undefined,
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#100F62",
          "borderRadius": 10,
          "elevation": 9,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.32,
          "shadowRadius": 5.46,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#070562",
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#869DB2",
              },
              Object {
                "fontFamily": "Nunito_600SemiBold",
                "fontSize": 17,
              },
              Object {
                "color": "#CEE1F2",
              },
              undefined,
            ]
          }
        >
          Alarm settings
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingVertical": 17,
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 17,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#869DB2",
                    },
                    Object {
                      "fontFamily": "Nunito_400Regular",
                      "fontSize": 17,
                    },
                    Object {
                      "color": "#CEE1F2",
                    },
                    undefined,
                  ]
                }
              >
                Progressive
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "#B1C2D1",
                "borderWidth": 0.5,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingVertical": 17,
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 17,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#869DB2",
                    },
                    Object {
                      "fontFamily": "Nunito_400Regular",
                      "fontSize": 17,
                    },
                    Object {
                      "color": "#CEE1F2",
                    },
                    undefined,
                  ]
                }
              >
                Adaptive
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "#B1C2D1",
                "borderWidth": 0.5,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingVertical": 17,
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 17,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#869DB2",
                    },
                    Object {
                      "fontFamily": "Nunito_400Regular",
                      "fontSize": 17,
                    },
                    Object {
                      "color": "#CEE1F2",
                    },
                    undefined,
                  ]
                }
              >
                Standard
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
